components/core/src/TimestampPattern.cpp

```diff
@@ -206,8 +210,9 @@ bool TimestampPattern::parse_timestamp (const string& line, epochtime_t& timesta
     const size_t format_length = m_format.length();
     size_t format_ix = 0;
     bool is_specifier = false;
+    bool is_relative = false;
```

Comment:
Since we now have 3 states (Unknown, IsSpecifier, IsRelativeFormat), can we use a State enum and change the ifs into a switch-case?

Reply:
commit 7b7a022ad396a29b675609a1f39633acd47218dc